docs/user_guides/fs/feature_view/query.md

```diff
@@ -75,35 +86,102 @@ By default, Hopsworks will use the maximal matching subset of the primary keys o
 Possible join types are "inner", "left" or "right". Furthermore, it is possible to specify different features for the join key of the left and right feature group.
 The join key lists should contain the names of the features to join on.
 
 === "Python"
     ```python
-    feature_join = rain_fg.select_all() \
-        .join(temperature_fg.select_all(), on=["date", "location_id"]) \
-        .join(location_fg.select_all(), left_on=["location_id"], right_on=["id"], join_type="left")
+    selected_features = credit_card_transactions_fg.select_all() \
+        .join(account_details_fg.select_all(), on=["cc_num"]) \
+        .join(merchant_details_fg.select_all(), left_on=["cc_num"], right_on=["merchant_id"], join_type="left")
     ```
 
 === "Scala"
     ```scala
-    val featureJoin = (rainFg.selectAll()
-        .join(temperatureFg.selectAll(), Seq("date", "location_id"))
-        .join(locationFg.selectAll(), Seq("location_id"), Seq("id"), "left"))
+    val selectedFeatures = (creditCardTransactionsFg.selectAll()
+        .join(accountDetailsFg.selectAll(), Seq("cc_num"))
+        .join(merchantDetailsFg.selectAll(), Seq("cc_num"), Seq("merchant_id"), "left"))
+    ```
+
+### Data modeling in Hopsworks
+
+Since v4.0 Hopsworks Feature selection API supports both Star and Snowflake Schema data models.
+
+#### Star schema data model
+
+When choosing Star Schema data model all tables are children of the parent (the left most) feature group, which has all 
+foreign keys for its children feature groups.
+
+<p align="center">
+  <figure>
+    <img src="../../../../assets/images/guides/fs/feature_view/star.png" alt="Star schema data model">
+    <figcaption>Star schema data model</figcaption>
+  </figure>
+</p>
+
+=== "Python"
+    ```python
+       selected_features = credit_card_transactions.select_all()
+        .join(aggregated_cc_transactions.select_all())
+        .join(account_details.select_all())
+        .join(merchant_details.select_all())
+        .join(cc_issuer_details.select_all())
     ```
 
-!!! error "Nested Joins"
-The API currently does not support nested joins. That is joins of joins.
-You can fall back to Spark DataFrames to cover these cases. However, if you have to use joins of joins, most likely there is potential to optimise your feature group structure.
+In online inference, when you want to retrieve features in your online model, you have to provide all foreign key values 
+from the parent feature group to retrieve your precomputed feature values using the feature view.
+
+=== "Python"
+    ```python
+      feature vector = feature_view.get_feature_vector({
+        ‘cc_num’: “1234 5555 3333 8888”,
+        ‘issuer_id’: 20440455,
+        ‘merchant_id’: 44208484,
+        ‘account_id’: 84403331
+        })
+    ```
+
+#### Snowflake schema 
+Hopsworks also provides the possibility to define children and grandchildren of the root (the left most) feature group. 
```

Comment:
Hopsworks also provides the possibility to define a feature view that consists of a nested tree of children (to up to a depth of 10) from the root (left most) feature group.

Reply:
Do we have a max depth for the tree?